core-js/src/main/javascript/cdf/components/filter/views/scrollbar/OptiScrollBarEngineImplementation.js

```diff
@@ -0,0 +1,41 @@
+/*!
+ * Copyright 2002 - 2016 Webdetails, a Pentaho company. All rights reserved.
+ *
+ * This software was developed by Webdetails and is provided under the terms
+ * of the Mozilla Public License, Version 2.0, or any later version. You may not use
+ * this file except in compliance with the license. If you need a copy of the license,
+ * please go to http://mozilla.org/MPL/2.0/. The Initial Developer is Webdetails.
+ *
+ * Software distributed under the Mozilla Public License is distributed on an "AS IS"
+ * basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. Please refer to
+ * the license for the specific language governing your rights and limitations.
+ */
+/**
+ * @summary OptiScrollBar implementation of ScrollBarHandler
+ * @description OptiScrollBar implementation of ScrollBarHandler
+ */
+define([
+  './AbstractScrollBarHandler',
+  '../../../../lib/jquery'
+],function(AbstractScrollBarHandler, $){
+  return AbstractScrollBarHandler.extend({
+    view: undefined,
+    scrollbar: undefined,
+    constructor: function (reference) {
+      this.view = reference;
+      this.scrollbar = this.view.$(this.view.config.view.slots.children).addClass('optiscroll-content').parent()
+            .addClass('optiscroll').optiscroll().off('scrollreachbottom').on('scrollreachbottom', function (event) {
+        return this.view.trigger('scroll:reached:bottom', this.view.model, event);
+      }).bind(this).off('scrollreachtop').on('scrollreachtop', function (event) {
```

Comment:
I see you tried to use `.bind(this)` in order to assign the correct `this` to the event handler, but this won't work, because `.bind(this)` returns the `on` method bound to the `OptiScrollBarEngineImplementation `, and not to the `optiscroll` object